### PR TITLE
lint code

### PR DESCRIPTION
**Pull request by stevenaldinger**
Hope this isn't aggressive in any way and I'm not explaining something you all already know but linting makes code more maintainable and easier for a community to get behind by adhering to certain stylistic conventions and can help catch bugs as well as make things very readable for people familiar with the language.

https://en.wikipedia.org/wiki/Lint_(software)

There's a project called [autopep8](https://github.com/hhatto/autopep8) that automatically runs `pylint` and fixes the formatting errors. 

`pip install --upgrade autopep8` installs it.

I wrote this command to go through and automatically update every .py file.

`find . -type f -name '*.py' -exec autopep8 --in-place --aggressive --aggressive {} \;`

### CODE REVIEW

**Comment by infrahumano** [comment]
<3

**Comment by XerxesNoble** [comment]
> Hope this isn’t aggressive in any way

‘—aggressive —aggressive’

**Comment by stevenaldinger** [comment]
> > Hope this isn’t aggressive in any way
> 
> ‘—aggressive —aggressive’

somehow didn't notice saying aggressive with aggressive in the command til later lol, I just copypasta'd that from their docs, looks like `--aggressive --aggressive` is probably accidental/redundant on their part but it did the job just fine.

**Comment by cclauss** [comment]
My vote would be for using __python/black__ instead of autopep8.  https://github.com/python/black

File: ehtim/__init__.py [hunk 1]
```diff
@@ -1,3 +1,26 @@
+from ehtim.const_def import *
+from ehtim.pipeline import Pipeline
+from ehtim.calibrating.network_cal import network_cal as netcal
+from ehtim.calibrating.self_cal import self_cal as selfcal
+from ehtim.calibrating.pol_cal import *
+from ehtim.plotting.summary_plots import *
+from ehtim.plotting.comparisons import *
+from ehtim.plotting.comp_plots import *
```
Comment (joseangel-sc) [inline ehtim/__init__.py]: Wild card

File: ehtim/__init__.py [hunk 2]
```diff
@@ -12,49 +35,28 @@
...
-from ehtim.calibrating import pol_cal
-from ehtim.plotting    import comp_plots
-from ehtim.plotting    import comparisons
 
-from ehtim.plotting.comp_plots import *
-from ehtim.plotting.comparisons import *
-from ehtim.plotting.summary_plots import *
-
-from ehtim.calibrating.pol_cal import *
 #from ehtim.calibrating.self_cal import *
```
Comment (joseangel-sc) [inline ehtim/__init__.py]: Remove unnecessary import with wild card

File: ehtim/array.py [hunk 3]
```diff
@@ -29,9 +29,11 @@
 from ehtim.const_def import *
```
Comment (joseangel-sc) [inline ehtim/array.py]: Wild card

File: ehtim/array.py [hunk 4]
```diff
@@ -101,20 +119,45 @@ def obsdata(self, ra, dec, rf, bw, tint, tadv, tstart, tstop, mjd=MJD_DEFAULT, t
...
-        scans = np.array([[time-0.5*tadv, time+0.5*tadv] for time in uniquetimes]) 
-        source=str(ra) + ":" + str(dec)
-        obs = ehtim.obsdata.Obsdata(ra, dec, rf, bw, obsarr, self.tarr, 
-                                    source=source, mjd=mjd, timetype=timetype, polrep=polrep,
-                                    ampcal=True, phasecal=True, opacitycal=True, 
-                                    dcal=True, frcal=True,
-                                    scantable=scans)
+        scans = np.array([[time - 0.5 * tadv, time + 0.5 * tadv]
+                          for time in uniquetimes])
+        source = str(ra) + ":" + str(dec)
```
Comment (joseangel-sc) [inline ehtim/array.py]:
```suggestion
        source = f"{a}:{dec}"
```
I think this is more readable

File: ehtim/array.py [hunk 5]
```diff
@@ -127,22 +170,24 @@ def make_subarray(self, sites):
...
 
     def save_txt(self, fname):
         """Save the array data in a text file.
 
            Args:
                fname (str) : path to output array file
         """
-        ehtim.io.save.save_array_txt(self,fname)
+        ehtim.io.save.save_array_txt(self, fname)
         return
```
Comment (joseangel-sc) [inline ehtim/array.py]: I know this has nothing to do with this PR, but what does this function return? 
```suggestion
       
```
IMO this is best

File: ehtim/features/rex.py [hunk 6]
```diff
@@ -223,97 +237,113 @@ def calc_meanprof_and_stats(self):
...
-        x0_c = mask.fovx()/2. - self.x0*eh.RADPERUAS
-        y0_c = self.y0*eh.RADPERUAS - mask.fovy()/2.
-        rad = self.RingSize1[0]*eh.RADPERUAS
+        x0_c = mask.fovx() / 2. - self.x0 * eh.RADPERUAS
+        y0_c = self.y0 * eh.RADPERUAS - mask.fovy() / 2.
+        rad = self.RingSize1[0] * eh.RADPERUAS
 
-        mask.imvec *= 0 
-        mask = mask.add_gauss(1,[2*rad,2*rad,0,x0_c,y0_c])
+        mask.imvec *= 0
```
Comment (joseangel-sc) [inline ehtim/features/rex.py]:
```suggestion
        mask.imvec = 0
```

File: ehtim/features/rex.py [hunk 7]
```diff
@@ -324,42 +354,52 @@ def calc_width(self, prof):
...
         pkrad, maxval = self.calc_pkrad_from_prof(prof)
-        spline = UnivariateSpline(self.rs, prof-0.5*maxval, s=0)
-        roots = spline.roots() # find the roots
+        spline = UnivariateSpline(self.rs, prof - 0.5 * maxval, s=0)
+        roots = spline.roots()  # find the roots
 
-        if len(roots)==0: 
-            return(self.rs[0],self.rs[-1])
+        if len(roots) == 0:
+            return(self.rs[0], self.rs[-1])
```
Comment (joseangel-sc) [inline ehtim/features/rex.py]:
```suggestion
            return self.rs[0], self.rs[-1]
```

File: ehtim/features/rex.py [hunk 8]
```diff
@@ -368,150 +408,196 @@ def plot_img(self, save_png=False):
                 subprocess.call(['mkdir', POSTPROCDIR + '/' + dirname])
 
             basename = os.path.basename(self.imname)
-            fname = POSTPROCDIR + '/' +dirname+ '/' + basename[:-5] + '_contour.png'
+            fname = POSTPROCDIR + '/' + dirname + \
```
Comment (joseangel-sc) [inline ehtim/features/rex.py]:
```suggestion
            fname = f"{POSTPROCDIR}/{dirname}\"
```

File: ehtim/features/rex.py [hunk 9]
```diff
@@ -368,150 +408,196 @@ def plot_img(self, save_png=False):
...
+        plt.tick_params(axis='both', which='major', length=6)
 
         if save_png:
             dirname = os.path.basename(os.path.dirname(self.imname))
             if not os.path.exists(POSTPROCDIR + '/' + dirname):
                 subprocess.call(['mkdir', POSTPROCDIR + '/' + dirname])
 
             basename = os.path.basename(self.imname)
-            fname = POSTPROCDIR + '/' +dirname+ '/' + basename[:-5] + '_unwrapped.png'
+            fname = POSTPROCDIR + '/' + dirname + \
```
Comment (joseangel-sc) [inline ehtim/features/rex.py]: I suggested this as a format string some line above, but even if that is not the chosen way to go, I prefer the old version better in a single line, this is not very readable

File: ehtim/io/writeData.py [hunk 10]
```diff
@@ -12,67 +12,150 @@
 import ehtim.io.oifits
 from ehtim.const_def import *
 
-def writeOIFITS(filename, RA, DEC, frequency, bandWidth, intTime,
-                visamp, visamperr, visphi, visphierr, u, v, ant1, ant2, timeobs,
-                t3amp, t3amperr, t3phi, t3phierr, uClosure, vClosure, antOrder, timeClosure,
-                antennaNames, antennaDiam, antennaX, antennaY, antennaZ):
 
+def writeOIFITS(
```
Comment (joseangel-sc) [inline ehtim/io/writeData.py]: definitely adds some extra lines that I don't find very pythonic

**Pull request closed without merging**